Update recommended renderer for XR (#10256)
godotengine/godot-docs: tutorials/rendering/renderers.rst

Choose **Mobile** if:

    - You are developing for newer mobile devices, desktop XR, or desktop.
    - You have relatively new hardware which supports Vulkan, Direct3D 12, or Metal.
    - You are developing a 3D game.
    - You want to use advanced rendering features, subject to the limitations
      of mobile hardware.

Choose **Compatibility** if:

    - You are developing for older mobile devices, older desktop devices, or

>       stand-alone XR. The Compatibility renderer supports the widest range of hardware.

Reviewer: ```suggestion
      standalone XR. The Compatibility renderer supports the widest range of hardware.
```
Consider "standalone" as one word instead of hyphenated. It's currently split within the docs and "standalone" seems common in general usage

|                     |                          |                          | optimized. Use Mobile or |
|                     |                          |                          | Compatibility instead.   |
+---------------------+--------------------------+--------------------------+--------------------------+

> | XR                  | ✔️ Yes. Recommended for  | ✔️ Yes. Recommended for | ⚠️ Supported, but poorly |

Reviewer: ```suggestion
| XR                  | ✔️ Yes. Recommended for  | ✔️ Yes. Recommended for  | ⚠️ Supported, but poorly |
```

> | XR                  | ✔️ Yes. Recommended for  | ✔️ Yes. Recommended for | ⚠️ Supported, but poorly |
> |                     | stand-alone headsets     |  desktop headsets        | optimized. Use Mobile or  |
> |                     |                          |                          | Compatibility instead.   |

Reviewer: ```suggestion
| XR                  | ✔️ Yes. Recommended for  | ✔️ Yes. Recommended for  | ⚠️ Supported, but poorly |
|                     | stand-alone headsets     |  desktop headsets        | optimized. Use Mobile or |
|                     |                          |                          | Compatibility instead.   |
```
The tables are misaligned. It's because emojis are annoyingly not monospaced, but a space and a half. I did my best guess to fix it with the web UI
